### PR TITLE
fix RNN-T loss bug - use logits instead of log_probs

### PR DESCRIPTION
**Pull request by aheba**
Hi @TParcollet ,
as reported in #1339, thanks to @csukuangfj,

Only torchaudio is using logits as input of the RNNT loss function. (other lib doesn't, see https://github.com/1ytic/warp-rnnt, https://github.com/HawkAaron/warp-transducer/tree/master/pytorch_binding)

Here is a fix to the RNN-T loss using torchaudio,

I will use this PR to add to our Numba code:
- [ ] // computation for alpha & beta.
- [ ] add fastemit_lambda (see https://arxiv.org/abs/2010.11148)

### CODE REVIEW

**Comment by csukuangfj** [comment]
> Only torchaudio is using logits as input of the RNNT loss function.

The advantage of using `logits` as input is that it can reduce memory consumption since you don't need to allocate space for `log_probs` and its gradients. Furthermore, the gradient of `logits` can share the same memory space as `logits`.

There are other implementations that take `logits` as input. See 
- https://github.com/danpovey/fast_rnnt/
- https://github.com/csukuangfj/optimized_transducer

**Comment by aheba** [comment]
Thanks for highlighting that point, let me take a look on :

https://github.com/csukuangfj/transducer-loss-benchmarking

**Comment by mravanelli** [comment]
@aheba, what's the status of this PR?

**Comment by mravanelli** [comment]
Thank you  @aheba  for the fix and @csukuangfj for the comments. I tested it and everything seems to work.
